jena-arq/src/main/java/org/apache/jena/riot/tokens/TokenizerText.java

```diff
@@ -60,6 +61,26 @@
     private boolean finished = false ;
     private TokenChecker checker = null ;
 
+    // The code assumes that errors throw exception and so stop parsing.
+    private ErrorHandler errorHandler = new ErrorHandler() {
```

Comment:
Will do. It is stateless though a stateless object is not very expensive to create!

The others are in ErrorHandlerFactory.

It maybe it becomes a parameter to the constructor.

Because letting dubious data through can lead to a lot of support load due to downstream database problems, this PR just adds mechanism and does not enable it. Hence it is keeping the default error handler in-file for now.